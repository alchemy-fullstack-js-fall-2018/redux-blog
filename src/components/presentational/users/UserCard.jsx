import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { ROUTES } from '../../../routes';
import styled from 'styled-components';

const StyledDiv = styled.div`
  margin: 5px;
  border: 1px solid black;
`;

const UserCard = ({ user }) => {
  const { id, name, email } = user;

  return (
    <StyledDiv>
      <p>id: {id}</p>
      <p>name: {name}</p>
      <p>email: {email}</p>
      <Link to={ROUTES.USER_DETAIL.linkTo(id)}>
        See more...
      </Link>
    </StyledDiv>
  );
};

UserCard.propTypes = {
  user: PropTypes.object.isRequired
};

export default UserCard;
