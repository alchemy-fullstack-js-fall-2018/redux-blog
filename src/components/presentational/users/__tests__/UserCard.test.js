import React from 'react';
import { shallow } from 'enzyme';
import UserCard from '../UserCard.jsx';
import mockUsers from '../../../../testing/fixtures/users.json';

describe('<UserCard />', () => {
  test('renders', () => {
    const user = mockUsers[0];
    const wrapper = shallow(
      <UserCard
        user={user}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
