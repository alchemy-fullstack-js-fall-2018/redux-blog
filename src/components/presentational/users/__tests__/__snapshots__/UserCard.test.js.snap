// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserCard /> renders 1`] = `
<styled.div>
  <p>
    id: 
    1
  </p>
  <p>
    name: 
    Leanne Graham
  </p>
  <p>
    email: 
    Sincere@april.biz
  </p>
  <Link
    replace={false}
    to="/users/1"
  >
    See more...
  </Link>
</styled.div>
`;
